ansible inventory groups not present in variables

Not sure if it's a bug or if it's related to the ansible_runner refact done in testinfra 3.0.2

Before 3.0.x release, we were collecting the ansible groups information from the inventory via the ansible variables :

```python
        ansible_vars = host.ansible.get_variables()
        num_mons = len(ansible_vars["groups"]["mons"])
```

In testinfra 2.1.0 the `ansible_vars["groups"]` is :
```console
{u'mgrs': [u'cephaio-1', u'cephaio-2', u'cephaio-3'], 'all': [u'cephaio-1', u'cephaio-2', u'cephaio-3'], u'clients': [u'cephaio-1', u'cephaio-2', u'cephaio-3'], u'osds': [u'cephaio-1', u'cephaio-2', u'cephaio-3'], 'ungrouped': [], u'mons': [u'cephaio-1', u'cephaio-2', u'cephaio-3'], u'mdss': [u'cephaio-1', u'cephaio-2', u'cephaio-3'], u'rgws': [u'cephaio-1', u'cephaio-2', u'cephaio-3']}
```

But with testinfra 3.0.x the groups key isn't defined anymore.
```python
        ansible_vars = host.ansible.get_variables()
>       num_mons = len(ansible_vars["groups"]["mons"])
E       KeyError: 'groups'
```

The ansible hosts file looks like :
```yaml
[mons]
cephaio-1
cephaio-2
cephaio-3

[mgrs]
cephaio-1
cephaio-2
cephaio-3

[osds]
cephaio-1
cephaio-2
cephaio-3

[rgws]
cephaio-1
cephaio-2
cephaio-3

[mdss]
cephaio-1
cephaio-2
cephaio-3

[clients]
cephaio-1
cephaio-2
cephaio-3
```

Is there another way to retrieve this information or is it a bug that breaks backward compatibility ?

--- a/test/test_backends.py
+++ b/test/test_backends.py
@@ -121,18 +121,25 @@ def test_ansible_get_variables():
 
         def get_vars(host):
             return AnsibleRunner(f.name).get_variables(host)
+        groups = {
+            'all': ['centos', 'debian'],
+            'g': ['debian'],
+            'ungrouped': ['centos'],
+        }
         assert get_vars("debian") == {
             'a': 'b',
             'c': 'd',
             'x': 'z',
             'inventory_hostname': 'debian',
             'group_names': ['g'],
+            'groups': groups,
         }
         assert get_vars("centos") == {
             'a': 'a',
             'e': 'f',
             'inventory_hostname': 'centos',
             'group_names': ['ungrouped'],
+            'groups': groups,
         }
 
 

--- a/test/test_modules.py
+++ b/test/test_modules.py
@@ -338,6 +338,11 @@ def test_ansible_module(host):
     assert variables["mygroupvar"] == "qux"
     assert variables["inventory_hostname"] == "debian_stretch"
     assert variables["group_names"] == ["testgroup"]
+    assert variables["groups"] == {
+        "all": ["debian_stretch"],
+        "testgroup": ["debian_stretch"],
+        "ungrouped": [],
+    }
 
     with pytest.raises(host.ansible.AnsibleException) as excinfo:
         host.ansible("command", "zzz")

--- a/testinfra/utils/ansible_runner.py
+++ b/testinfra/utils/ansible_runner.py
@@ -142,16 +142,21 @@ class AnsibleRunner(object):
 
     def get_variables(self, host):
         inventory = self.inventory
+        # inventory_hostname, group_names and groups are for backward
+        # compatibility with testinfra 2.X
         hostvars = inventory['_meta'].get(
             'hostvars', {}).get(host, {})
         hostvars.setdefault('inventory_hostname', host)
-        groups = []
+        group_names = []
+        groups = {}
         for group in sorted(inventory):
-            if group in ('_meta', 'all'):
+            if group == "_meta":
                 continue
-            if host in inventory[group].get('hosts', []):
-                groups.append(group)
-        hostvars.setdefault('group_names', groups)
+            groups[group] = sorted(list(itergroup(inventory, group)))
+            if group != "all" and host in inventory[group].get('hosts', []):
+                group_names.append(group)
+        hostvars.setdefault('group_names', group_names)
+        hostvars.setdefault('groups', groups)
         return hostvars
 
     def get_host(self, host, **kwargs):

